test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageFileServiceTests.cs

```diff
@@ -153,6 +193,45 @@ public async Task GetPackageIconAsync_EmbeddedFromFallbackFolder_DoesNotLockFile
             }
         }
 
+        [Fact]
+        public async Task GetPackageIconAsync_NonEmbedded_GetStream_DoesNotLockFile()
+        {
+            // Arrange
+            using (TestDirectory testDirectory = TestDirectory.Create())
+            {
+                string fileContents = "testFileContents";
+                string filePath = Path.Combine(testDirectory.Path, "testFile.txt");
+
+                File.WriteAllText(filePath, fileContents);
+
+                var packageFileService = new NuGetPackageFileService(
+                        default(ServiceActivationOptions),
+                        Mock.Of<IServiceBroker>(),
+                        new AuthorizationServiceClient(Mock.Of<IAuthorizationService>()),
+                        _telemetryProvider.Object);
+                var packageIdentity = new PackageIdentity(nameof(GetPackageIconAsync_NonEmbedded_GetStream_DoesNotLockFile), NuGetVersion.Parse("1.0.0"));
+
+                // Do not add a fragment which is typically required for the embedded icon.
+                // Note: file:// is required for Uri to parse the Fragment (possibly a Uri bug).
+                var uri = new Uri("file://" + filePath);
+                NuGetPackageFileService.AddIconToCache(packageIdentity, uri);
+                FileStream fileNotLocked = File.Open(uri.LocalPath, FileMode.Open, FileAccess.ReadWrite, FileShare.Read);
```

Comment:
I don't understand the scenario for this test. When would Visual Studio need to read a file from the global packages folder that is concurrently being written to by another file handle?

NuGet doesn't handle files in either the global packages folder, or file feed sources, changing during the process (which is a problem since we have a button to clear the global packages folder, but this test isn't testing delete, it's testing write).  I imagine this test isn't intended to test reading the icon while the nupkg is still getting extracted. I'm not sure that package search would find the package in such a scenario.

Reply:
As I stated in the `Secondary refactoring` section of this PR description, I also don't think VS needs to write these files, yet if you look at the linked bug - VS was _opening_ the files for Write. The test is proving that we're not opening the icon for Write and only for Read.

Reply:
I've changed the tests and refactored variable/test names to try to be more clear about testing that PM UI can Read a file, even if already open for Read, but will not read the file if it's open for Write.

Let me know if this looks better.